### DISCUSSION TITLE
Option to add additional auth headers?

### DISCUSSION THREAD

**Discussion by qLutCh55** [dedoc/scramble#363] [Q&A] [answered]
Hello,

I have looked through the documentation, and can't seem to find anywhere where you can define custom headers?

Currently our api is secured using a combination of custom headers namely 'x-api-key' and 'x-api-secret'. I am able to add a single header with the open the following code:

```
Scramble::extendOpenApi(function (OpenApi $openApi) {
        $openApi->secure(
            SecurityScheme::apiKey('header', 'x-api-key')
        );
    });
```

but I need to be able to add a second header 'x-api-secret'.

Any suggestions?

Thanks

**Comment by romalytvynenko** [comment]
Hey @qLutCh55 

Any chance you can show me the expected OpenAPI document for your auth setup? Based on that I will be able to help with Scramble's setup.

**Comment by qLutCh55** [comment]
Hey @romalytvynenko 

I was trying to get something like this right:

![image](https://github.com/dedoc/scramble/assets/26224124/38d9f284-c70e-4182-a6b7-a4dd005be7c0)

or

![image](https://github.com/dedoc/scramble/assets/26224124/2b92c75f-ff7d-4f10-8a7a-bb6ed7b6fc94)

But the idea is that I need more than 1 security header for requests. With the above code, it only takes the first header and ignores the second one.

Regards
Jeandrew

**Reply by romalytvynenko** [reply]
@qLutCh55 gotcha! 

Are you sure this is supported by Open API specification itself?

**Comment by qLutCh55** [comment]
I'm not sure if it is supported by Open API specification itself. It just seemed to be the closest implementation I could find in the docs for security for the purpose of our project. If you could point me in another direction, I would greatly appreciate it.

**Reply by romalytvynenko** [reply]
@qLutCh55 the question here is Open API spec supports such kind of security. If it does, I'm more than open to add it to the Scramble! Otherwise UI won't be able to reflect this even if added, as UI renders only what is supported by the spec.

**Reply by lord-obinna** [reply]
i also need this

**Reply by romalytvynenko** [reply]
@lord-obinna the question remains: is it supported by OpenAPI? If so, can you share the example. If not, I won't be able to help.

**Reply by giggsey** [reply]
https://swagger.io/docs/specification/authentication/

Isn't it the 'Applying security' supporting multiple:
```
security:
  - ApiKeyAuth: []
  - OAuth2:
      - read
      - write
      - ```

**Reply by romalytvynenko** [reply]
@giggsey the issue here is that the auth is done by 2 headers at the same time (you need to pass 2 values to be authenticated) – if I understood correctly. And I did't see this is supported by OpenAPI spec.

**Reply by giggsey** [reply]
@romalytvynenko Yup, my mistake. I came here looking for multiple auth types in general (e.g. query OR bearer). I'll keep searching and raise a discussion if required.

**Comment by romalytvynenko** [comment, marked as answer]
@giggsey @lord-obinna @qLutCh55 

Scramble now (v0.12.7) supports this!

https://scramble.dedoc.co/usage/authentication#complex-authentication-scenario
